src/components/TeacherComponents/OnboardingForm/OnboardingForm.tsx

```diff
@@ -217,6 +195,7 @@ const OnboardingForm = ({
               type="submit"
               onClick={props.onSubmit}
               aria-description={submitError ?? undefined}
+              isLoading={isSubmitting}
```

Comment:
Setting the form button to loading mode while the async actions take place to prevent users from spamming the button